contracts/vault/src/execute.rs

```diff
@@ -18,6 +23,30 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> R
     match msg {
         ExecuteMsg::Deposit {} => execute_deposit(deps, env, info),
 
+        ExecuteMsg::Receive(Cw20ReceiveMsg {
+            sender,
+            amount,
+            msg: _,
```

Comment:
The implication here is that all CW20 messages will only be an `ExecuteMsg::Deposit`. IMO it's still worth parsing and confirming that to avoid unintentional loss of funds.